src/lib/drawer/Drawer.svelte

```diff
@@ -1,43 +1,43 @@
 <script lang="ts">
-	import type { SvelteHTMLElements } from 'svelte/elements';
 	import { twMerge } from 'tailwind-merge';
-	import Button from '$lib/Button.svelte';
 	import { fly } from 'svelte/transition';
+	import { clickOutside } from './clickOutside.js';
 
-	type $$Props = SvelteHTMLElements['div'] & { open?: boolean };
-
-	let className: $$Props['class'] = undefined;
+	let className = '';
 	export { className as class };
+	export let overlayClass = '';
 	export let open = false;
+	export let disableClickOutside = false;
 
-	function closeDrawer() {
-		open = false;
+	function handleClickOutside(): void {
+		if (!disableClickOutside) {
+			open = false;
+		}
 	}
 </script>
 
 {#if open}
 	<div
 		tabindex="-1"
-		transition:fly={{ x: '100%', duration: 300 }}
+		transition:fly={{ x: '100%', duration: 200 }}
```

Comment:
this transition probably wants moving on the the drawer itself rather than the overlay